osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmapSet.cs

```diff
@@ -111,6 +111,30 @@ private void load(BeatmapManager manager, BeatmapSetOverlay beatmapOverlay)
             };
         }
 
+        private readonly List<DrawableCarouselBeatmap> beatmapsToCheck = new List<DrawableCarouselBeatmap>();
+
+        public void AssureBeatmapPositionValid(DrawableCarouselBeatmap beatmap) => beatmapsToCheck.Add(beatmap);
+
+        public void ClearAssurance() => beatmapsToCheck.Clear();
+
+        protected override void Update()
+        {
+            base.Update();
+
+            var i = 0;
+
+            beatmapsToCheck.ForEach(beatmap =>
+            {
+                if (beatmap.Y > Y + DrawHeight + 6 + i * (beatmap.DrawHeight + 5))
+                    beatmap.MoveToY(Y + DrawHeight + 5 + i * (beatmap.DrawHeight + 5)).MoveToY(beatmap.DesiredY, 750, Easing.OutExpo);
+
+                if (beatmap.Y < Y + DrawHeight + 4)
+                    beatmap.MoveToY(Y + DrawHeight + 5).MoveToY(beatmap.DesiredY, 750, Easing.OutExpo);
```

Comment:
Why `> Y + DrawHeight + 6` and `< Y + DrawHeight + 4`? Those look like off-by-ones (I would have expected 5 in both places? Keep in mind it's strict inequality)

Reply:
I logged the positions of beatmap panels when they were triggering these checks. They seemed to be needlessly hitting them when these were set to 5 so the off-by-ones are intentionally here.

Reply:
Needlessly how? If it's due to floating point then I'd use something from `Precision` to have a check with tolerance. The way this is currently written looks buggy.

Reply:
I don't think it's due to floating point. I think the easing is just different enough (each panel starts from different position so their velocity will also differ) that the difference between Y positions will just grow larger until it will hit any ``acceptableDifference`` threshold. Debating removing the check altogether if the animation doesn't suffer from it.